### PR TITLE
resolves #394 fix gem requirements resolution

### LINKED ISSUE #394: [2.0.0-alpha.3] Installs a different version of a gem specified
```gradle
buildscript {
	repositories {
		maven {
			url 'https://dl.bintray.com/jruby-gradle/plugins'
		}
	}
}

plugins {
	id 'com.github.jruby-gradle.base' version '2.0.0-alpha.3'
}

repositories {
	jcenter()
	ruby.gems()
}

dependencies {
	gems 'rubygems:selenium-webdriver:3.142.6'
	gems 'rubygems:webdrivers:4.1.3'
}
```

Running `gradle jrubyPrepare` on this emits the following output:

```
Successfully installed mini_portile2-2.4.0
Successfully installed nokogiri-1.10.7-java
Successfully installed rubyzip-2.0.0
Successfully installed childprocess-3.0.0
Successfully installed selenium-webdriver-4.0.0.alpha3
Successfully installed webdrivers-4.1.3
```

As you can see, the installed version of `selenium-webdriver` is `4.0.0.alpha3` while `3.142.6` is requested. This totally breaks the usage of that gem.

### PR DESCRIPTION
**Pull request by Mogztter**
resolves #394

### CODE REVIEW

File: core-plugin/src/test/groovy/com/github/jrubygradle/api/gems/GemVersionSpec.groovy [hunk 1]
```diff
@@ -42,15 +41,16 @@ class GemVersionSpec extends Specification {
...
+        gemRequirement  | ivyNotation
+        '= 1.0.0'       | '1.0.0'
+        '> 2.0'         | ']2.0,)'
+        '>= 2.2.0'      | '[2.2.0,)'
+        '<= 3.0'        | '(,3.0]'
+        '< 2.3.0'       | '(,2.3.0['
+        '~> 1.0'        | '[1.0.0,2.0['
+        '~> 2.2'        | '[2.2.0,3.0['
+        '~> 2.2.0'      | '[2.2.0,2.3.0['
+        '>= 3.0, < 4.0' | '[3.0, < 4.0,)'
```
Comment (Mogztter) [inline core-plugin/src/test/groovy/com/github/jrubygradle/api/gems/GemVersionSpec.groovy]: not sure about the trailing comma?

File: core-plugin/src/main/groovy/com/github/jrubygradle/api/gems/GemVersion.groovy [hunk 2]
```diff
@@ -300,66 +300,53 @@ class GemVersion implements Comparable<GemVersion> {
...
-                newLowBoundary = other.lowBoundary
-                break
-            case 0:
-                newLowBoundary = (lowBoundary == EXCLUSIVE || other.lowBoundary == EXCLUSIVE) ? EXCLUSIVE : INCLUSIVE
-                newLow = low
-                break
-            case 1:
-                newLow = low
-                newLowBoundary = lowBoundary
+        Tuple2<String, Boundary> newLowVersionSpec = intersect(low, lowBoundary, other.low, other.lowBoundary, true)
```
Comment (Mogztter) [inline core-plugin/src/main/groovy/com/github/jrubygradle/api/gems/GemVersion.groovy]: We should probably introduce a data class to encapsulate the version and the boundary:

```groovy
class GemVersion {
    final GemVersionConstraint low
    final GemVersionConstraint high
}

class GemVersionConstraint {
  final String version
  final String boundary
}
```

But I don't want to introduce too much changes in a bug-fix pull request.

**Comment by Mogztter** [comment]
Build is green! Review is highly appreciated :wink: 

ping @ysb33r

File: core-plugin/src/main/groovy/com/github/jrubygradle/api/gems/GemVersion.groovy (hunk 2 above)
Comment (ysb33r) [reply core-plugin/src/main/groovy/com/github/jrubygradle/api/gems/GemVersion.groovy]: Feel free to do so in a subsequent pR.

**Pull request merged**